src/App.js

```diff
@@ -1,91 +1,111 @@
+import { Switch, Route, useParams } from 'react-router-dom'
 import ProductList from './components/ProductList/ProductList'
-import burgers from './mocks/burger.json'
-import kebabs from './mocks/kebab.json'
-import minced from './mocks/minced.json'
-import nuggets from './mocks/nuggets.json'
-import sausages from './mocks/sausages.json'
-import productCategories from './mocks/category.json'
+import useFetch from '../src/hooks/useFetch'
 import { ReactComponent as Logo } from './images/icons/logo.svg'
 import styled from 'styled-components'
 import Headline from './components/Headline/Headline'
+import Detailpage from './components/Detailpage/Detailpage'
 
 export default function App() {
+  const burgers = useFetch('burger')
+  const kebabs = useFetch('kebab')
+  const minced = useFetch('minced')
+  const nuggets = useFetch('nuggets')
+  const sausages = useFetch('sausages')
+  const categories = useFetch('categories')
+
   return (
     <Switch>
       <Route exact path="/">
         <LogoStyled />
-        {productCategories.map((category, index) => (
+        {categories.map((category, index) => (
           <ProductOverview
             key={index}
             productIcon={category.icon}
             productName={category.name}
           />
         ))}
       </Route>
-      <Route path="/burger">
-        <>
-          <Headline headline="Burger" />
-          {burgers.map((burger, index) => (
-            <ProductList
-              title={burger.title}
-              image={burger.image}
-              key={index}
-            />
-          ))}
-        </>
+      <Route exact path="/burger">
+        <Headline headline="Burger" />
+        {burgers.map((burger, index) => (
+          <ProductList
+            title={burger.title}
+            image={burger.image}
+            key={index}
+            id={burger.id}
+            category={'burger'}
+          />
+        ))}
       </Route>
-      <Route path="/cold-cuts">
-        <>
-          <Headline headline="Cold-Cuts" />
-          <SubheadlineStyled>Coming soon</SubheadlineStyled>
-        </>
+      <Route path={'/burger/:id'}>
+        <Detailpage category={'burger'} />
       </Route>
-      <Route path="/kebab">
-        <>
-          <Headline headline="Kebab" />
-          {kebabs.map((kebab, index) => (
-            <ProductList title={kebab.title} image={kebab.image} key={index} />
-          ))}
-        </>
+      <Route exact path="/cold-cuts">
+        <Headline headline="Cold-Cuts" />
+        <SubheadlineStyled>Coming soon</SubheadlineStyled>
       </Route>
-      <Route path="/minced">
-        <>
-          <Headline headline="Minced" />
-          {minced.map((minced, index) => (
-            <ProductList
-              title={minced.title}
-              image={minced.image}
-              key={index}
-            />
-          ))}
-        </>
+      <Route exact path="/kebab">
+        <Headline headline="Kebab" />
```

Comment:
This and the next lines are almost identical to the `/burger` route. How about making this a component with props `title` and `products` ... and `category` (maybe you don't even need that last one). You could use it in all similar routes.